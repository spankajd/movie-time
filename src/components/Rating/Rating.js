import React from 'react';
import './Rating.scss';

const Rating = ({rating}) => {
    const totalRating = 10;
    const EmptyStar = '/assets/Empty_Star.png';
    const FullStar = '/assets/Full_Star_Yellow.png';
        
    const starArray = [];

    for(let i = 0; i < totalRating; i++) {
        starArray.push(<img 
            className="star"
            src={ i < rating ? FullStar : EmptyStar} key={i}
            alt={ i < rating ? "full star" : "empty star"}
        />);
    }

    return <div className='rating' >
        {rating ? starArray : 'N/A'}
    </div>
};

export default Rating;