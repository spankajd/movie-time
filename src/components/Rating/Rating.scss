.rating {
    display: inline-flex;
    justify-content: space-between;
    align-items: center;
    width: 150px;
    padding: 3px;
    .star {
        flex: 1;
        width: 10px;
    }
}
