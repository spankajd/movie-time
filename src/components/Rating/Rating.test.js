import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import Rating from "./Rating";

describe("Rating Component", () => {
    it("renders Rating", () => {
      render(<Rating rating={null} />);
      const element = screen.getByText(/N\/A/i);
      expect(element).toBeInTheDocument();
    });

    it("matches snapshot", () => {
        const { asFragment } = render(<Rating />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders Rating with 5 star", () => {
        render(<Rating rating={5} />);
        const element = screen.getAllByAltText(/full star/i)[0];
        expect(element).toBeInTheDocument();
      });
      
});