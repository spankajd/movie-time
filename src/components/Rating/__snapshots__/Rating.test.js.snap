// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rating"
  >
    N/A
  </div>
</DocumentFragment>
`;
